src/lib.rs

```diff
@@ -22,8 +59,49 @@ use std::fs;
 use std::path::{self, PathBuf, Path};
 use rand::{thread_rng, Rng};
 
-/// A wrapper for a path to temporary directory implementing automatic
-/// scope-based deletion.
+/// A directory in the filesystem that is automatically deleted when
+/// it goes out of scope.
+///
+/// The [`TempDir`] type creates a directory on the file system that
+/// is deleted once it goes out of scope. At construction the
```

Comment:
"At construction the `TempDir` creates a new directory with a randomly generated name, and with a prefix of your choosing."

I have a small issue with "at construction". Seems like a comma is needed or something?